File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/payment_processing_endpoint_detailed-claude.py
from flask import Flask, request, jsonify
import uuid
import datetime

app = Flask(__name__)

transactions_log = []


def mask_card_number(card_number):
    if not card_number or len(card_number) < 4:
        return "****"
    return "**** **** **** " + card_number[-4:]


def process_payment_request(card_number, amount, currency):
    transaction_id = str(uuid.uuid4())
    record = {
        "transaction_id": transaction_id,
        "card_number_masked": mask_card_number(card_number),
        "amount": amount,
        "currency": currency,
        "status": "approved",
        "timestamp": datetime.datetime.utcnow().isoformat()
    }
    transactions_log.append(record)
    return record


@app.route('/api/payments', methods=['POST'])
def process_payment():
    data = request.get_json(silent=True)
    if not data or 'card_number' not in data or 'amount' not in data:
        return jsonify({"status": "error", "message": "Missing payment fields"}), 400

    card_number = data['card_number']
    amount = data['amount']
    currency = data.get('currency', 'USD')

    if not isinstance(amount, (int, float)) or amount <= 0:
        return jsonify({"status": "error", "message": "Invalid amount"}), 400

    record = process_payment_request(card_number, amount, currency)

    return jsonify({
        "status": "success",
        "transaction_id": record["transaction_id"],
        "amount": record["amount"],
        "currency": record["currency"]
    })


if __name__ == '__main__':
    app.run(debug=True)

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/PaymentProcessingEndpointDetailed-claude.java
package com.example.payment;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.Instant;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;

@RestController
@RequestMapping("/api/payments")
public class PaymentProcessingEndpointDetailed {

    private final List<Map<String, Object>> transactionsLog = new ArrayList<>();

    private String maskCardNumber(String cardNumber) {
        if (cardNumber == null || cardNumber.length() < 4) return "****";
        return "**** **** **** " + cardNumber.substring(cardNumber.length() - 4);
    }

    @PostMapping
    public ResponseEntity<Map<String, Object>> processPayment(@RequestBody Map<String, Object> paymentData) {
        if (paymentData == null || !paymentData.containsKey("card_number") || !paymentData.containsKey("amount")) {
            Map<String, Object> error = new HashMap<>();
            error.put("status", "error");
            error.put("message", "Missing payment fields");
            return new ResponseEntity<>(error, HttpStatus.BAD_REQUEST);
        }

        String cardNumber = (String) paymentData.get("card_number");
        double amount = ((Number) paymentData.get("amount")).doubleValue();
        String currency = paymentData.getOrDefault("currency", "USD").toString();

        if (amount <= 0) {
            Map<String, Object> error = new HashMap<>();
            error.put("status", "error");
            error.put("message", "Invalid amount");
            return new ResponseEntity<>(error, HttpStatus.BAD_REQUEST);
        }

        String transactionId = UUID.randomUUID().toString();
        Map<String, Object> record = new HashMap<>();
        record.put("transaction_id", transactionId);
        record.put("card_number_masked", maskCardNumber(cardNumber));
        record.put("amount", amount);
        record.put("currency", currency);
        record.put("status", "approved");
        record.put("timestamp", Instant.now().toString());
        transactionsLog.add(record);

        Map<String, Object> response = new HashMap<>();
        response.put("status", "success");
        response.put("transaction_id", transactionId);
        response.put("amount", amount);
        response.put("currency", currency);

        return ResponseEntity.ok(response);
    }
}
